benchmarks/hash_table/static_multimap/find_all_bench.cu

```diff
@@ -0,0 +1,152 @@
+std::enable_if_t<(sizeof(Key) == sizeof(Value)), void> nvbench_find_all(
+  nvbench::state& state,
+  nvbench::type_list<Key, Value, nvbench::enum_type<CGSize>, nvbench::enum_type<BufferSize>>)
+{
+  std::size_t const num_keys = state.get_int64("NumInputs");
+  auto const occupancy       = state.get_float64("Occupancy");
+  std::size_t const size     = num_keys / occupancy;
+  std::size_t const num_reps = state.get_int64("NumReps");
+
+  constexpr bool is_outer = true;
+
+  std::vector<Key> h_keys(num_keys);
+  std::vector<cuco::pair_type<Key, Value>> h_pairs(num_keys);
+
+  generate_multikeys<Key>(h_keys.begin(), h_keys.end(), num_reps);
+
+  for (auto i = 0; i < num_keys; ++i) {
+    Key key           = h_keys[i];
+    Value val         = h_keys[i];
+    h_pairs[i].first  = key;
+    h_pairs[i].second = val;
+  }
+
+  // Get an array of unique keys
+  std::set<Key> key_set(h_keys.begin(), h_keys.end());
+  std::vector<Key> h_unique_keys(key_set.begin(), key_set.end());
+  thrust::device_vector<Key> d_unique_keys(h_unique_keys);
+
+  thrust::device_vector<cuco::pair_type<Key, Value>> d_results(2 * num_keys);
+  thrust::device_vector<cuco::pair_type<Key, Value>> d_pairs(h_pairs);
+
+  state.add_element_count(num_keys, "NumKeys");
+  state.add_global_memory_writes<Key>(num_keys * 2);
+
+  state.exec(
+    nvbench::exec_tag::sync | nvbench::exec_tag::timer, [&](nvbench::launch& launch, auto& timer) {
+      cuco::static_multimap<Key, Value, cuco::double_hashing<Key, Value, CGSize>> map{size, -1, -1};
+      map.insert(d_pairs.begin(), d_pairs.end());
+
+      timer.start();
+      auto view = map.get_device_view();
+
+      auto const block_size  = 128;
+      auto const warp_size   = 32;
+      auto const buffer_size = CGSize * BufferSize;
+      auto const stride      = 1;
+      auto const grid_size = (CGSize * num_keys + stride * block_size - 1) / (stride * block_size);
+
+      using KeyEqual = thrust::equal_to<Key>;
+
+      KeyEqual key_equal;
+
+      using atomic_ctr_type = typename cuco::static_multimap<Key, Value>::atomic_ctr_type;
+      atomic_ctr_type* num_items;
+      CUCO_CUDA_TRY(cudaMallocManaged(&num_items, sizeof(atomic_ctr_type)));
+      *num_items = 0;
+      int device_id;
+      CUCO_CUDA_TRY(cudaGetDevice(&device_id));
+      CUCO_CUDA_TRY(cudaMemPrefetchAsync(num_items, sizeof(atomic_ctr_type), device_id));
+
+      // Use timers to explicitly mark the target region
+      cuco::detail::vectorized_retrieve<block_size, warp_size, CGSize, buffer_size, is_outer>
+        <<<grid_size, block_size, 0, launch.get_stream()>>>(d_unique_keys.begin(),
+                                                            d_unique_keys.end(),
+                                                            d_results.data().get(),
+                                                            num_items,
+                                                            view,
+                                                            key_equal);
```

Comment:
Shouldn't we only be benchmarking the public interface? Remember that this has bitten us before.